sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveSessionCatalog.scala

```diff
@@ -55,16 +55,19 @@ private[sql] class HiveSessionCatalog(
     conf,
     hadoopConf) {
 
-  override def lookupRelation(name: TableIdentifier, alias: Option[String]): LogicalPlan = {
+  override def lookupRelation(
+      name: TableIdentifier,
+      alias: Option[String],
+      databaseHint: Option[String] = None): LogicalPlan = {
     val table = formatTableName(name.table)
-    val db = formatDatabaseName(name.database.getOrElse(currentDb))
+    val db = formatDatabaseName(name.database.getOrElse(databaseHint.getOrElse(currentDb)))
     if (db == globalTempViewManager.database) {
       val relationAlias = alias.getOrElse(table)
       globalTempViewManager.get(table).map { viewDef =>
         SubqueryAlias(relationAlias, viewDef, Some(name))
       }.getOrElse(throw new NoSuchTableException(db, table))
     } else if (name.database.isDefined || !tempTables.contains(table)) {
-      val database = name.database.map(formatDatabaseName)
+      val database = Some(db).map(formatDatabaseName)
```

Comment:
-> `val database = formatDatabaseName(db)`

Then, you can change the following line to Some(database)